File: backend/src/main/java/com/back/ourlog/domain/genre/entity/Genre.java
package com.back.ourlog.domain.genre.entity;

import jakarta.persistence.*;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@NoArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class Genre {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(unique = true)
    private String name;

    @OneToMany(mappedBy = "genre", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<DiaryGenre> diaryGenres = new ArrayList<>();

    public Genre(String name) {
        this.name = name;
    }

}

File: backend/src/main/java/com/back/ourlog/domain/genre/entity/Genre.kt
package com.back.ourlog.domain.genre.entity

import jakarta.persistence.*

@Entity
class Genre(

    @Column(unique = true, nullable = false)
    var name: String

) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int? = null

    @OneToMany(
        mappedBy = "genre",
        cascade = [CascadeType.ALL],
        orphanRemoval = true,
        fetch = FetchType.LAZY
    )
    val diaryGenres: MutableList<DiaryGenre> = mutableListOf()

    override fun equals(other: Any?): Boolean =
        this === other || (other is Genre && id != null && id == other.id)

    override fun hashCode(): Int = id?.hashCode() ?: 0
}
